llama_index/vector_stores/types.py

```diff
@@ -65,23 +65,47 @@ class ExactMatchFilter(BaseModel):
     value: Union[StrictInt, StrictFloat, StrictStr]
 
 
+class AdvancedFilter(BaseModel):
+    """Advanced metadata filter for vector stores to support more operators.
+
+    Value uses Strict* types, as int, float and str are compatible types and were all
+    converted to string before.
+
+    See: https://docs.pydantic.dev/latest/usage/types/#strict-types
+    """
+
+    key: str
+    value: Union[StrictInt, StrictFloat, StrictStr]
+    operator: Optional[str] = None
```

Comment:
wouldn't this technically deprecate `ExactMatchFilter` ?

I would prefer an enum for the operator

Reply:
It seems like right now, this requires knowledge about the operators for the specific vector store -- it would be nice to have unified interface for all vector stores I think?

Reply:
if we use Enum will it be limited to some operators. different vector store have different operators. everytime a new operation come in. we will need to add one by one right?

Reply:
I think there's only so many operators: 

`<`, `>`, `<=`, `>=`, `==` and `!=`

Then, each vector store should need to translate that into the appropriate syntax (or raise an error if not supported)

This is similar to the current `ExactMatchFilter` (i.e. `==`) where the filter is translated into the required syntax for each vector store